From 7329589ce3183792477308ea9f040ab38cd52b5a Mon Sep 17 00:00:00 2001
From: Anirudh Ghayal <aghayal@codeaurora.org>
Date: Thu, 14 May 2015 23:40:26 +0530
Subject: [PATCH 295/959] power: qpnp-smbcharger-mmi: Fix uninitialized access
 to dc_psy

Fix uninitialized access when dc_psy is not registered
and charging is enabled by sysfs interface.

CRs-Fixed: 836629
Change-Id: I88f1f2253c357a62589ea750a4394ef289f5573e
Signed-off-by: Anirudh Ghayal <aghayal@codeaurora.org>
---
 drivers/power/qpnp-smbcharger-mmi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/qpnp-smbcharger-mmi.c b/drivers/power/qpnp-smbcharger-mmi.c
index 9529e41a14c..a017cb5e9cd 100644
--- a/drivers/power/qpnp-smbcharger-mmi.c
+++ b/drivers/power/qpnp-smbcharger-mmi.c
@@ -1515,7 +1515,7 @@ static int smbchg_dc_en(struct smbchg_chip *chip, bool enable,
 		goto out;
 	}
 
-	if (chip->psy_registered)
+	if (chip->dc_psy_type != -EINVAL)
 		power_supply_changed(&chip->dc_psy);
 	pr_smb(PR_STATUS, "dc charging %s, suspended = %02x\n",
 			suspended == 0 ? "enabled"
-- 
2.11.0

